kong/plugins/cors/handler.lua

```diff
@@ -25,21 +25,26 @@ local function configure_origin(ngx, conf)
   if #conf.origins == 1 then
     if conf.origins[1] == "*" then
       ngx.ctx.cors_allow_all = true
+      ngx.header["Access-Control-Allow-Origin"] = "*"
+      return
 
     else
       ngx.header["Vary"] = "Origin"
     end
 
-    ngx.header["Access-Control-Allow-Origin"] = conf.origins[1]
-    return
+    -- if this doesnt look like a regex, set the ACAO header directly
+    -- otherwise, we'll fall through to an iterative search and
+    -- set the ACAO header based on the client Origin
+    if re_find(conf.origins[1], "^[A-Za-z0-9.:/-]+$", "jo") then
```

Comment:
I think we should log possible errors here, just as a standardized practice among our codebase when using the `ngx.re.*` API (I believe other places do it, or should).